core/system/src/pool_queue.c

```diff
@@ -100,8 +100,9 @@ static int enqueue(uvisor_pool_queue_t * pool_queue, uvisor_pool_slot_t slot)
     }
     uvisor_pool_queue_entry_t * slot_entry = &pool->management_array[slot];
 
-    if (slot_entry->dequeued.state == UVISOR_POOL_SLOT_IS_FREE) {
-        /* Reject if a free slot is being enqueued */
+    if (slot_entry->dequeued.state != UVISOR_POOL_SLOT_IS_DEQUEUED) {
```

Comment:
Consider checking for each state explicitly, in case more states are added in the future.